eva_submission/eload_validation.py

```diff
@@ -353,6 +358,93 @@ def _collect_validation_workflow_results(self, output_dir):
+            validator_output_files.append(os.path.join(self._get_dir('vcf_check'), vcf_name + '.vcf_validator.txt'))
+        has_sv = self._detect_structural_variant_from_validator_output(has_sv, validator_output_files)
+        for index in range(len(vcf_files)):
+            self.eload_cfg.set('validation', 'structural_variant_check', 'files', os.path.basename(vcf_files[index]),
+                               value={'has_structural_variant': has_sv[index]})
+        self.eload_cfg.set('validation', 'structural_variant_check', 'pass', value=True)
+        return has_sv
+
+    def _detect_structural_variant_from_variant_lines(self, vcf_files):
+        # Initializing the list of boolean values to store the status of the VCFs denoting whether it has a SV or not
+        has_sv = []
+        no_of_variant_lines = []
+        vcf_count = -1
+        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Pages: 6, 16)
+        symbolic_allele_pattern = re.compile("^<(DEL|INS|DUP|INV|CNV|BND|DUP:TANDEM|DEL:ME.*|INS:ME.*)>$")
+        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Page: 17)
+        complex_rearrangements_breakend_pattern = re.compile("^[ATCGNatgcn]+\[.+:.+\[$|^[ATCGNatgcn]+\].+:.+\]$|^\].+:.+\][ATCGNatgcn]+$|^\[.+:.+\[[ATCGNatgcn]+$")
+        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Page: 18)
+        complex_rearrangements_special_breakend_pattern = re.compile("^[ATGCNatgcn]+<[0-9A-Za-z!#$%&+./:;?@^_|~-][0-9A-Za-z!#$%&*+./:;=?@^_|~-]*>$")
+        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Page: 22)
+        single_breakend_pattern = re.compile("^\.[ATGCNatgcn]+|[ATGCNatgcn]+\.$")
+        for vcf_file in vcf_files:
+            vcf_count = vcf_count + 1
+            no_of_variant_lines_per_vcf = 0
+            has_sv_per_vcf = False
+            if vcf_file.endswith('.vcf.gz'):
+                open_file = gzip.open(vcf_file, mode="rt")
+            else:
+                open_file = open(vcf_file, mode="r")
+            for file_line in open_file:
+                if file_line[0] == "#":
+                    continue;
+                no_of_variant_lines_per_vcf = no_of_variant_lines_per_vcf + 1
+                if no_of_variant_lines_per_vcf > 10000:
+                    break
+                extract_columns = file_line.split("\t")
+                alt_allele_column = extract_columns[4]
+                alternate_alleles = alt_allele_column.split(",")
+                for alternate_allele in alternate_alleles:
+                    if re.search(symbolic_allele_pattern, alternate_allele) or \
+                            re.search(complex_rearrangements_breakend_pattern, alternate_allele) \
+                            or re.search(complex_rearrangements_special_breakend_pattern,
+                                         alternate_allele) or re.search(single_breakend_pattern, alternate_allele):
+                        has_sv_per_vcf = True
```

Comment:
**Optional:** actually report the specific type of SV that we found...

Reply:
**Optional:** Even more useful is to actually collect these variants along with the SV types in a file for further reference.

Reply:
Yes. I will implement the reporting of the SVs as well in a different ticket